pkg/api/api_test.go

```diff
@@ -0,0 +1,61 @@
+package api
+
+import (
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/require"
+	"github.com/weaveworks/common/server"
+)
+
+type FakeLogger struct{}
+
+func (fl *FakeLogger) Log(keyvals ...interface{}) error {
+	return nil
+}
+
+func TestNewApiWithoutSourceIPExtractor(t *testing.T) {
```

Comment:
This test is a bit weak. It doesn't really assert on the functionality, but just that the `New()` doesn't return error.

_Same comment applies to `TestNewApiWithSourceIPExtractor`._

Reply:
Yeah, I was hoping to piggy back on some existing API test but there was nothing there. At least this shows the code doesn't panic. I haven't looked at the integration tests yet in case . Will do that next.